### PR TITLE
Merge plugin version 2019072601 (Release v3.3)

### PR DESCRIPTION
**Pull request by paulsteffen**
Add privacy api provider (Issue #2)
Fixe names of db tables to match the coding gudelines (Issue #3)
Move reading of db records from constructor of class basepost to child classes, use hard-coded tablenames there (Issue #4)
Remove orphaned files (Issue #6)
Add backup api support (Issue #7)
Schedule task at random time(Issue #8)
Tidy up language strings (Issue #9)
Create instance comment only after require_login check (Issue #10)
Prefix css definitions (Issue #11)
Fixed code formatting, added more inline documentation (Issue #12)

### CODE REVIEW

File: db/tasks.php [hunk 1]
```diff
@@ -15,16 +15,21 @@
...
+defined('MOODLE_INTERNAL') || die();
+
 $tasks = array(
     array(
         'classname' => 'block_socialcomments\task\process_digest_cron',
         'blocking' => 0,
-        'minute' => '0',
-        'hour' => '0',
+        'minute' => 'R',
+        'hour' => 'R',
```
Comment (paulsteffen) [inline db/tasks.php:32]: Resolve #8

File: classes/local/basepost.php [hunk 2]
```diff
@@ -39,20 +42,7 @@ abstract class basepost {
...
-            if ($dbattrs = $DB->get_record($this->tablename, array('id' => $attrs['id']), '*', $strictness)) {
-
-                // Load new content, if available.
-                if (isset($attrs['content'])) {
-                    $dbattrs->content = $attrs['content'];
-                }
-
-                $attrs = (array) $dbattrs;
-            }
-        }
```
Comment (paulsteffen) [inline classes/local/basepost.php:55]: Code moved to constuctor of each child class (comment, reply).

File: classes/local/comment.php [hunk 3]
```diff
@@ -45,8 +45,21 @@ class comment extends basepost {
...
+                // Load new content, if available.
+                if (isset($attrs['content'])) {
+                    $dbattrs->content = $attrs['content'];
+                }
+
+                $attrs = (array) $dbattrs;
+            }
+        }
 
-        $this->tablename = 'block_scomments_comments';
```
Comment (paulsteffen) [inline classes/local/comment.php:49]: $DB->get_record is now called with a hard-coded table name.

File: classes/local/reply.php [hunk 4]
```diff
@@ -40,8 +42,21 @@ class reply extends basepost {
...
+                // Load new content, if available.
+                if (isset($attrs['content'])) {
+                    $dbattrs->content = $attrs['content'];
+                }
+
+                $attrs = (array) $dbattrs;
+            }
+        }
 
-        $this->tablename = 'block_scomments_replies';
```
Comment (paulsteffen) [inline classes/local/reply.php:44]: $DB->get_record is now called with a hard-coded table name.

**Comment by annebardtke** [comment]
@BjoernLefers @paulsteffen @Elearning-HWR-Berlin 
I'll merge this as suggested by Dan Marsden in https://tracker.moodle.org/browse/CONTRIB-7463?focusedCommentId=786557&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-786557

**Pull request merged**